plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts

```diff
@@ -52,7 +52,7 @@ export interface useCollectionOptions {
 export const useCollection = < T >(
 	options: useCollectionOptions
 ): {
-	results: T[];
+	results: T;
```

Comment:
Is there a particular reason for this change? Reading the JSDoc, it looks like that `results` is always an array:

https://github.com/woocommerce/woocommerce/blob/a4b04feeca676b97144e33339eac5b8f52d54aca/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L37-L41

Reply:
> Is there a particular reason for this change?

Yes, the `results` is not always an array. 


<img width="464" alt="Screenshot 2024-10-31 at 5 30 28 AM" src="https://github.com/user-attachments/assets/a515c8b1-55d5-480a-9791-8204c69c26aa">

<img width="298" alt="image" src="https://github.com/user-attachments/assets/ef853e11-4874-46db-ae37-724e1ba9d29b">

> Reading the JSDoc, it looks like that results is always an array:

So, the assumption that results always return an array is wrong, and the documentation is also incorrect. I'm going to fix it.

This led us to use the [unknown](https://github.com/woocommerce/woocommerce/blob/ac9dbb2a2f4d8dc93dff11abd423320d5d81e8d6/plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/edit.tsx#L90) word here, @LuigiPulcini. Why are you using this here?


<img width="309" alt="image" src="https://github.com/user-attachments/assets/7d18b260-24d9-4b92-a5e6-ec3b5cfae76f">